### PR TITLE
docs: update prose about image tag/name format

### PR DESCRIPTION
**Pull request by dvdksn**
**- What I did**

Update the verbiage about the format of image tags/names.

- Addresses docker/docs#20893

**- How I did it**

/usr/bin/nvim

**- How to verify it**

read it

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by laurazard — APPROVED** [review]
This looks fantastic! Thanks 🤗

**Review by vvoland — APPROVED** [review]

File: docs/reference/commandline/image_tag.md [hunk 1]
```diff
@@ -12,38 +12,50 @@ Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE
...
-
-After the image name, the optional `TAG` is a custom, human-readable manifest
-identifier that's typically a specific version or variant of an image. The tag
-must be valid ASCII and can contain lowercase and uppercase letters, digits,
-underscores, periods, and hyphens. It can't start with a period or hyphen and
-must be no longer than 128 characters. If you don't specify a tag, the command uses `latest` by default.
-
-You can group your images together using names and tags, and then
-[push](image_push.md) them to a registry.
+A Docker image name consists of several components that describe where the
```
Comment (vvoland) [inline docs/reference/commandline/image_tag.md:15]:
```suggestion
A Docker image reference consists of several components that describe where the
```

File: docs/reference/commandline/image_tag.md [hunk 2]
```diff
@@ -12,38 +12,50 @@ Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE
...
+: The namespace (optional) usually represents a user or organization. The
+  repository is required and identifies the specific image. If the namespace is
+  omitted, Docker defaults to `library`, the namespace reserved for Docker
+  Official Images.
+
+`TAG`
+: An optional identifier used to specify a particular version or variant of the
+  image. If no tag is provided, Docker defaults to `latest`.
+
+### Example image names
```
Comment (vvoland) [inline docs/reference/commandline/image_tag.md:39]:
```suggestion
### Example image references
```

**Review by vvoland — APPROVED** [review]
LGTM, thanks!

**Pull request merged**